### PR TITLE
Port math.rs to sqllogictest

### PR DESCRIPTION
**Pull request by 2010YOUY01**
# Which issue does this PR close?

Part of https://github.com/apache/arrow-datafusion/issues/4495.

# Rationale for this change

Ported tests in math.rs into sqllogictest.
The engine for sqllogictest will round big decimals to 12 digits, so the output of test cases is also round to 12 digits.
# What changes are included in this PR?

# Are these changes tested?

# Are there any user-facing changes?

### CODE REVIEW

**Review by jiangzhx — APPROVED** [review]

**Review by waynexia — APPROVED** [review]
Looks good to me 👍

**Review by r4ntix — APPROVED** [review]

File: datafusion/core/tests/sqllogictests/test_files/math.slt [hunk 1]
```diff
@@ -87,3 +87,24 @@ query RRRRRRRR
 SELECT round(125.2345, -3), round(125.2345, -2), round(125.2345, -1), round(125.2345), round(125.2345, 0), round(125.2345, 1), round(125.2345, 2), round(125.2345, 3)
 ----
 0 100 130 125 125 125.2 125.23 125.235
+
+# Create table for atan2 test
+statement ok
```
Comment (comphead) [inline datafusion/core/tests/sqllogictests/test_files/math.slt:92]: its good to drop table after the test completed, or as alternative you can prepare data in WITH clause.

Comment (2010YOUY01) [reply datafusion/core/tests/sqllogictests/test_files/math.slt]: Thank you for the feedback, I think even creating a table is not necessary. Updated

File: datafusion/core/tests/sqllogictests/test_files/math.slt [hunk 2]
```diff
@@ -87,3 +87,9 @@ query RRRRRRRR
 SELECT round(125.2345, -3), round(125.2345, -2), round(125.2345, -1), round(125.2345), round(125.2345, 0), round(125.2345, 1), round(125.2345, 2), round(125.2345, 3)
 ----
 0 100 130 125 125 125.2 125.23 125.235
+
+# atan2
+query RRRR
```
Comment (comphead) [inline datafusion/core/tests/sqllogictests/test_files/math.slt:92]: please include also tests with nulls. Its likely gap from prev implementation
`atan2(null, 1.0), atan2(2.0, null)`

Comment (2010YOUY01) [reply datafusion/core/tests/sqllogictests/test_files/math.slt]: Sure, expression tests for null cases can also be found [here](https://github.com/apache/arrow-datafusion/pull/2942/files#diff-d6e2f75f82c6d7f14f1bb07bc89293c7ef4c92c4c02241044e14ad9ea8ae213dR508), returning `NULL` seems to be expected

**Review by comphead — APPROVED** [review]
cool, thanks @2010YOUY01

**Comment by Weijun-H** [comment]
LGTM

**Comment by alamb** [comment]
Thank you @2010YOUY01 and everyone who reviewed this!

**Pull request merged**